cli/src/cmd/forge/script/broadcast.rs

```diff
@@ -0,0 +1,266 @@
+        self.wait_for_receipts(receipts, deployment_sequence).await?;
+
+        println!("\n\n==========================");
+        println!(
+            "\nONCHAIN EXECUTION COMPLETE & SUCCESSFUL. Transaction receipts written to {:?}",
+            deployment_sequence.path
+        );
+        Ok(())
+    }
+
+    pub async fn send_transaction(
+        &self,
+        tx: TypedTransaction,
+        signer: SignerMiddleware<Provider<Http>, Wallet<SigningKey>>,
+        wait: bool,
+        chain: u64,
+        is_legacy: bool,
+        fork_url: &str,
+    ) -> eyre::Result<(TransactionReceipt, U256)> {
+        let mut legacy_or_1559 = if is_legacy {
+            TypedTransaction::Legacy(tx.into())
+        } else {
+            TypedTransaction::Eip1559(tx.into())
+        };
+        legacy_or_1559.set_chain_id(chain);
+
+        let from = *legacy_or_1559.from().expect("no sender");
+
+        if wait {
+            match foundry_utils::next_nonce(from, fork_url, None) {
+                Ok(nonce) => {
+                    let tx_nonce = *legacy_or_1559.nonce().expect("no nonce");
+
+                    if nonce != tx_nonce {
+                        eyre::bail!("EOA nonce changed unexpectedly while sending transactions.");
+                    }
+                }
+                Err(_) => {
+                    eyre::bail!("Not able to query the EOA nonce.");
+                }
+            }
+        }
+
+        async fn broadcast<T, U>(
+            signer: SignerMiddleware<T, U>,
+            legacy_or_1559: TypedTransaction,
+        ) -> eyre::Result<Option<TransactionReceipt>>
+        where
+            SignerMiddleware<T, U>: Middleware,
+        {
+            tracing::debug!("sending transaction: {:?}", legacy_or_1559);
+            match signer.send_transaction(legacy_or_1559, None).await {
+                Ok(pending) => pending.await.map_err(|e| eyre::eyre!(e)),
+                Err(e) => Err(eyre::eyre!(e.to_string())),
+            }
+        }
+
+        let nonce = *legacy_or_1559.nonce().expect("no nonce");
+        let receipt = match broadcast(signer, legacy_or_1559).await {
+            Ok(Some(res)) => (res, nonce),
+
+            Ok(None) => {
+                // todo what if it has been actually sent
+                eyre::bail!("Failed to get transaction receipt?")
+            }
+            Err(e) => {
+                eyre::bail!("Aborting! A transaction failed to send: {:#?}", e)
+            }
+        };
+
+        Ok(receipt)
+    }
+
+    pub async fn handle_broadcastable_transactions(
+        &self,
+        target: &ArtifactId,
+        result: ScriptResult,
+        decoder: &mut CallTraceDecoder,
+        script_config: &ScriptConfig,
+    ) -> eyre::Result<()> {
+        if let Some(txs) = result.transactions {
+            if script_config.evm_opts.fork_url.is_some() {
+                if let Ok(gas_filled_txs) =
+                    self.execute_transactions(txs, script_config, decoder).await
```

Comment:
let's turn all the else branches that are `eyre::bail!` either into `.map_err(eyre::..)?` or `.wrap_err()?` followed by the ? operator